testng-core/src/main/java/org/testng/internal/invokers/BaseInvoker.java

```diff
@@ -54,7 +55,12 @@ protected void runInvokedMethodListeners(
 
     InvokedMethodListenerInvoker invoker =
         new InvokedMethodListenerInvoker(listenerMethod, testResult, testResult.getTestContext());
-    for (IInvokedMethodListener currentListener : m_invokedMethodListeners) {
+    boolean isBeforeInvocation = InvokedMethodListenerMethod.BEFORE_INVOCATION == listenerMethod;
```

Comment:
Could you add a comment which explains why we have a specific behavior for `BEFORE_INVOCATION`?

Reply:
For `BEFORE_INVOCATION` method, still run as insert order, but regarding `AFTER_INVOCATION`, it should be reverse order.

Reply:
Why not, but why is it important? How/where can we track the rule?

Could you implement what you describe by replacing `isBeforeInvocation` with `isAfterInvocation`?